### DISCUSSION TITLE
Endpoint to dispatch OAuth flow providing an `access_token`

### DISCUSSION THREAD

**Discussion by XaviTorello** [fastapi-users/fastapi-users#658] [Ideas]
Hi dudes ✋🏼 !

We're reviewing how can we integrate OAuth flows with an externally-generated `access_token`.

The goal is to be able to POST to a new endpoint that receives the `access_token` and requests the validation to the provider and follows the same logic than the `/authorize` flow.

This will provide us the possibility to extract OS `access_tokens` and submit it to the backend to avoid unnecessary web authentications on the provider.

As an example, this will provide a simple way to integrate OAuth to our iOS/Android apps without the need to dispatch a browser-based authentication.

We're prototyping it, if it sounds ok we can submit a PR.

Many thanks!

**Comment by frankie567** [comment]
I'm not sure to understand everything but if I'm reading you correctly, you are able to retrieve OAuth access_tokens from iOS/Android OS without going through the authorization flow?

That's sound weird to me. I'm not expert in mobile development, but I guess that mobile SDK ends up giving you the `authorization_code` ; which then you can feed up to a backend server (in FastAPI Users, the `/callback` route) to exchange a proper access token: https://developers.google.com/identity/protocols/oauth2/native-app

If you really need an endpoint to do what you need, I guess you can implement it quite easily by taking [parts of the `/callback` logic](https://github.com/frankie567/fastapi-users/blob/9e35dbf450363c2d931113230ac813ace399ef6d/fastapi_users/router/oauth.py#L90-L156); but again, I'm not sure where you want to go here.
